### PR TITLE
Add doxygen into gf_getfld.f, Part of #55

### PR DESCRIPTION
**Pull request by Hang-Lei-NOAA**
Part of #55
This has been tested on
https://hang-lei-noaa.github.io/NCEPLIBS/NCEPLIBS-g2/html/files.html

### CODE REVIEW

File: src/gf_getfld.f [hunk 1]
```diff
@@ -1,229 +1,196 @@
+!>    @file
+!>    @brief Contains subroutines returns the Grid Definition,
+!>    and Product Definition for a given data field.
+!>    @author Stephen Gilbert @date 2000-05-26
+!>
+
+!>    This subroutine returns the Grid Definition, Product Definition,
+!>    Bit-map ( if applicable ), and the unpacked data for a given data
```
Comment (edwardhartnett) [inline src/gf_getfld.f]: Remove extra space inside the parens, so:

 `Bit-map (if applicable), and the unpacked data for a given data
`

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: src/gf_getfld.f (hunk 1 above)
Comment (Hang-Lei-NOAA) [reply src/gf_getfld.f]: Thanks

File: src/gf_getfld.f [hunk 2]
```diff
@@ -1,229 +1,196 @@
...
+!>   
+!>    PROGRAM HISTORY LOG:
+!>    2000-05-26 Stephen Gilbert
+!>    2002-01-24 Stephen Gilbert Changed to pass back derived type gribfield
+!>    variable through argument list, instead of having many different arguments.
+!>    2004-05-20 Stephen Gilbert Added check to see if previous a bit-map is
+!>    specified, but none was found.
+!>    2015-10-29 Boi Vuong Initial all pointers in derive type gribfield.
+!>
+!>    @param[in] cgrib Character array that contains the GRIB2 message
```
Comment (BrianCurtis-NOAA) [inline src/gf_getfld.f]: period at end

File: src/gf_getfld.f [hunk 3]
```diff
@@ -1,229 +1,196 @@
...
+!>    - 9 Data Representation Template 5.NN not yet implemented.
+!>    - 10 Error unpacking Section 3.
+!>    - 11 Error unpacking Section 4.
+!>    - 12 Error unpacking Section 5.
+!>    - 13 Error unpacking Section 6.
+!>    - 14 Error unpacking Section 7.
+!>    - 17 Previous bitmap specified, but none exists.
+!>
+!>    @note Note that derived type gribfield contains pointers to many
+!>    arrays of data.  The memory for these arrays is allocated when the
```
Comment (BrianCurtis-NOAA) [inline src/gf_getfld.f]: Only one space after period.

File: src/gf_getfld.f [hunk 4]
```diff
@@ -1,229 +1,196 @@
...
+!>    values in the arrays are set, to help minimize problems with array
+!>    overloading. Because of this users are encouraged to free up this memory,
+!>    when it is no longer needed, by an explicit call to subroutine gf_free.
+!>    Subroutine gb_info can be used to first determine how many data fields
+!>    exist in a given GRIB message.It may not always be possible to expand 
+!>    a bit-mapped data field. If a pre-defined bit-map is used and not 
+!>    included in the GRIB2 message itself, this routine would not have the 
+!>    necessary information to expand the data.  In this case, gfld\%expanded
```
Comment (BrianCurtis-NOAA) [inline src/gf_getfld.f]: Only one space after period

File: src/gf_getfld.f [hunk 5]
```diff
@@ -1,229 +1,196 @@
...
+!>    type variable, gfld. Gfld is of type gribfield, which is defined
+!>    in module grib_mod, so users of this routine will need to include
+!>    the line "USE GRIB_MOD" in their calling routine. Each component of the
+!>    gribfield type is described in the OUTPUT ARGUMENT LIST section below.
+!>    Since there can be multiple data fields packed into a GRIB2
+!>    message, the calling routine indicates which field is being requested
+!>    with the ifldnum argument.
+!>   
+!>    PROGRAM HISTORY LOG:
+!>    2000-05-26 Stephen Gilbert
```
Comment (edwardhartnett) [inline src/gf_getfld.f]: Make these into a table by adding '- ' before each entry.

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: src/gf_getfld.f (hunk 5 above)
Comment (Hang-Lei-NOAA) [reply src/gf_getfld.f]: Thanks

File: src/gf_getfld.f (hunk 4 above)
Comment (Hang-Lei-NOAA) [reply src/gf_getfld.f]: Thanks

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by edwardhartnett — APPROVED** [review]

**Pull request merged**